tests/Aspire.Hosting.Kafka.Tests/KafkaFunctionalTests.cs

```diff
@@ -101,12 +102,32 @@ public async Task WithDataShouldPersistStateBetweenUsages(bool useVolume)
             else
             {
                 bindMountPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+
+                if (!Directory.Exists(bindMountPath))
+                {
+                    if (OperatingSystem.IsWindows())
+                    {
+                        Directory.CreateDirectory(bindMountPath);
+                    }
+                    else
+                    {
+                        // the docker container runs as a non-root user, so we need to grant other user's read/write permission
+                        // to the bind mount directory.
+                        const UnixFileMode BindMountPermissions =
+                            UnixFileMode.UserRead | UnixFileMode.UserWrite | UnixFileMode.UserExecute |
+                            UnixFileMode.OtherRead | UnixFileMode.OtherWrite | UnixFileMode.OtherExecute;
+
+                        Directory.CreateDirectory(bindMountPath, BindMountPermissions);
```

Comment:
The permissions have to happen after the create, due to `umask`. See 

https://github.com/dotnet/aspire/pull/5087/files#diff-faa1118f611b9b62c2f9a92ef4498a54aac418a5c8e569d9f5565b9d6353e473R90-R102